addons/promote/common/src/main/java/org/commonjava/indy/promote/validate/PromotionValidationTools.java

```diff
@@ -561,6 +563,66 @@ public Transfer getTransfer( final StoreResource resource )
         runParallelAndWait( entries, closure, logger );
     }
 
+    public <T> void paralleledInBatch( Collection<T> collection, int batchSize, Closure closure )
```

Comment:
I think we should consider using an Indy configuration point for the batch size, instead of giving control over that to the script. Scripts are a hotbed source of bugs IMO, and keeping them consistent is much harder than managing the Indy configurations.

Reply:
@jdcasey Fair enough. Will go update it. 
BTW, I came up another theory about the promotion timeout. It looks promotion is not just slow. I suspect some tasks were blocked. Looking at no-pre-existing-paths, there are nested paralleledEach . Might be the higher level tasks saturated the threads pool so the lower level tasks never get a chance to run? and the higher level tasks waiting and timeout.

Reply:
Yep, I think removing the inner parallel execution is probably what we want. It's the smaller loop, for most builds, which gives us more control in our batching.